vllm/spec_decode/top1_proposer.py

```diff
@@ -73,6 +73,29 @@ def get_proposals(
                 execute_model_req=nonzero_execute_model_req,
                 sample_len=proposal_len,
             )
+            if maybe_sampler_output is not None:
+                # Some sequences do not get speculative tokens from
+                # the draft worker. Remove these sequences from nonzero
+                # proposal len seqs to reduce scoring overhead
+                zero_seq_idxs = []
+                for seq_idx, sampler_output in zip(
+                        nonzero_proposal_len_indices, maybe_sampler_output):
+                    if sampler_output is None:
+                        proposal_lens[seq_idx] = 0
+                        zero_seq_idxs.append(seq_idx)
+                nonzero_proposal_len_indices = [
+                    idx for idx in nonzero_proposal_len_indices
+                    if idx not in zero_seq_idxs
```

Comment:
* If we skip a sequence in this case, does it still do normal decoding?
* Can we add a test that hits this path? lazily, we could do an ngram e2e test over sufficiently long generation. more precisely, we could do a unit test on top1 proposer or ngram worker.

Reply:
* It should just do the same thing as other sequences with zero proposal lengths (due to max context length or auto-disabling). Plz correct if I'm wrong.
* An existing test already covers this case: https://github.com/vllm-project/vllm/pull/4724/files#diff-4ce22933dc7ebf3c65c3b299481cfafe4e3493a37dd32c5bf4df54d0dc70d18bL66. The first sequence was having `proposals.proposal_lens[0] = proposal_len` and `proposals.proposal_token_ids[0] = [0,0,...]`, and now having `proposals.proposal_lens[0]=0` and `proposals.proposal_token_ids[0] = [-1,-1,...]`.

Reply:
> It should just do the same thing as other sequences with zero proposal lengths (due to max context length or auto-disabling). Plz correct if I'm wrong.

I follow now, sounds good!

> An existing test already covers this case:

Great!